### PR TITLE
Closes #7 create test nv suppnv and ag for neuro

### LINKED ISSUE #7: Test Data: NV, SUPPNV and AG
### Background Information

Add NV or XP test dataset (decide on naming!) to {admiralneuro}, to contain amyloid/tau PET imaging data

Note 1: this issue should be on hold until #6 is completed. Once we have a DM dataset, ensure to use the same patients for NV/XP.

Note 2: this test data will be moved to [{pharmaversesdtm}](https://github.com/pharmaverse/pharmaversesdtm/) after the first release of {admiralneuro}

### Definition of Done

- [ ] `data-raw` folder should contain new program `nv_neuro` that creates DM datasetDM test dataset added to {admiralneuro} package
 - [ ] `R/data.R` should be updated to document the new DM dataset (see [here](https://github.com/pharmaverse/admiral/blob/main/R/data.R) for an example file)

### PR DESCRIPTION
**Pull request by jwang-lilly**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral family codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiral<ext> (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the Reference page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Review by manciniedoardo — CHANGES_REQUESTED** [review]
Thanks @jwang-lilly! Looking much better. I left a comment around simplifying your code. Also I have the following general comments (some may be due to my unfamiliarity with the TA).

- NVTESTCD = "FTPSICBGM" is still >8 chars long
- NVTESTCD = "FBPVR" only has "Positive" as results - is that intended? 
- NVTESTCD = "FBVPR" and "FTPVR" - is it correct we only have baseline records?

I leave @pharmaverse/admiralneuro to review the actual data.

**Comment by Fanny-Gautier** [comment]
Should we add the Reference Region in `SUPPNV` as suggested by Chris in #7 ?
image

**Comment by Fanny-Gautier** [comment]
If we want to add `AG` domain with the Tracer info, we need to add `NVLNKID`in our `NV` domain.
image

File: data-raw/nv_neuro.R [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+# Load libraries ----
+
+library(tibble)
+library(dplyr)
+library(stringr)
+library(admiral)
+
+# Read input data ----
+
+data("dm_neuro")
```
Comment (Fanny-Gautier) [inline data-raw/nv_neuro.R:13]: Note: It will have to be replaced by `dm_neuro <- pharmaversesdtm::dm_neuro` once it is added in `pharmaversesdtm` (after first release only)

**Review by Fanny-Gautier — COMMENTED** [review]
`NVLNKID` to add, and `Reference Region` to probably add in `SUPPNV`.
1 Note left to read `dm_neuro` after the first release of `{admiralneuro}`

**Comment by jwang-lilly** [comment]
@Fanny-Gautier, @cgpele, any particular reason 
image

**Comment by jwang-lilly** [comment]
> If we want to add `AG` domain with the Tracer info, we need to add `NVLNKID` in our `NV` domain. image image

I think the scanners should have been harmonized using phantom images, so at the stages of SDTM or ADaM, there shouldn't be a need to account for differences among scanners.

**Comment by cgpele** [comment]
For the tracer information, if I read the SDTM IG correctly, this should also go elsewhere (in AG) - but again I will defer to @manciniedoardo on how aligned our SDTM need to be.  Thinking about the fuction for calculating the centiloids, perhaps it would be more transparent if we seperate out all of the factors needed into their own variables, i.e. for tracer, pipeline, regions etc?

**Comment by Fanny-Gautier** [comment]
Hi @jwang-lilly ,
Here is the way `AG` and `NV` are linked (using `xxLNKID`) together as per TAUG. Also giving you an example of `AG` domain.
image

**Comment by jwang-lilly** [comment]
Tracers are added to AG.  NV is simplified with links to AG.  SUPPNV is used for PET SUVR reference regions.  Thanks everyone for great suggestions.

**Comment by manciniedoardo** [comment]
@cgpele @Fanny-Gautier how do @jwang-lilly 's updates look to you?

File: data-raw/ag_neuro.R [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+# Read input data ----
+
+data("nv_neuro")
+
+# Convert blank to NA ----
+
+nv_neuro <- convert_blanks_to_na(nv_neuro)
+
+ag_neuro <- nv_neuro %>%
+  dplyr::mutate(DOMAIN = "AG", AGSEQ = NVSEQ, AGLNKID = NVLNKID, AGSTDTC = NVDTC) %>%
```
Comment (Fanny-Gautier) [inline data-raw/ag_neuro.R:20]: Patient 01-701-1015 (as for example) is having `NV.NVLNKID` from 1 to 5, all on '2014-01-02'.
Corresponding links are created in `AG` as `AGLNKID`. Does it mean that the patient received 5 different times an Agent (twice 18F-Florbetapir, once 18F-Florbetaben, and twice 18F-Flortaucipir) on the same date? or should the `xxLNKID` be grouped by agent administration? e.g having only 3 records in `AG` and corresponding `xxLNKID` such as `xxLNKID = 1` for 18F-Florbetapir, `xxLNKID = 2` for 18F-Florbetaben and `xxLNKID = 3`  for 18F-Flortaucipir.

File: data-raw/suppnv_neuro.R [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  QVAL = "Qualifier Value"
+)
+
+for (var in names(labels)) {
+  attr(suppnv_neuro[[var]], "label") <- labels[[var]]
+}
+
+# Label SUPPNV dataset ----
+
+attr(suppnv_neuro, "label") <- "Supplemental to Nervous System Findings"
```
Comment (Fanny-Gautier) [inline data-raw/suppnv_neuro.R:48]: I think should be "Supplemental **Qualifiers for** Nervous ..."

File: data-raw/ag_neuro.R [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+  AGSTDTC = "Date/Time of Administration"
+)
+
+for (var in names(labels)) {
+  attr(ag_neuro[[var]], "label") <- labels[[var]]
+}
+
+# Label AG dataset ----
+
+attr(ag_neuro, "label") <- "Procedure Agents for Nervous System"
```
Comment (Fanny-Gautier) [inline data-raw/ag_neuro.R:68]: Should be "Procedure Agents" only

File: data-raw/suppnv_neuro.R [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+labels <- list(
+  STUDYID = "Study Identifier",
+  RDOMAIN = "Related Domain Abbreviation",
+  USUBJID = "Unique Subject Identifier",
+  IDVAR = "Identifying Variable",
+  IDVARVAL = "Identifying Variable Value",
+  QNAM = "Qualifier Variable Name",
+  QLABEL = "Qualifier Variable Label",
+  QVAL = "Qualifier Value"
```
Comment (Fanny-Gautier) [inline data-raw/suppnv_neuro.R:39]: should be "Data Value"

File: data-raw/suppnv_neuro.R [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+)
```
Comment (Fanny-Gautier) [inline data-raw/suppnv_neuro.R:40]: `QORIG` ("Origin") variable is missing. To be added - Required variable.
Depending on the values of `QORIG`, the variable `QEVAL` might also be needed.

**Review by Fanny-Gautier — CHANGES_REQUESTED** [review]
I think that's it from my end. Available to discuss about the `xxLNKID` issue, or if someone else has any other approach, please feel free to add your suggestion.

File: data-raw/nv_neuro.R [hunk 7]
```diff
@@ -0,0 +1,248 @@
...
+
+# Add labels to variables ----
+
+labels <- list(
+  # Identifier Variables (Key variables)
+  STUDYID = "Study Identifier",
+  DOMAIN = "Domain Abbreviation",
+  USUBJID = "Unique Subject Identifier",
+  NVSEQ = "Sequence Number",
+  NVLNKID = "Link to Subject's AG Domain Record",
```
[the remark this answers is not in the document]
Comment (jwang-lilly) [reply data-raw/nv_neuro.R:207]: Good, changed to Link ID.

File: data-raw/nv_neuro.R [hunk 8]
```diff
@@ -0,0 +1,248 @@
...
+
+  # Topic Variables\
+  NVTESTCD = "Nervous System Test Short Name",
```
[the remark this answers is not in the document]
Comment (jwang-lilly) [reply data-raw/nv_neuro.R:210]: Ok, changed to "Short Name of Nervous System Test".

File: data-raw/nv_neuro.R [hunk 9]
```diff
@@ -0,0 +1,248 @@
...
+  NVTEST = "Nervous System Test Name",
+  NVCAT = "Category for Nervous System Test",
+
+  # Qualifier Variables
+  NVLOC = "Location of Nervous System Finding",
```
[the remark this answers is not in the document]
Comment (jwang-lilly) [reply data-raw/nv_neuro.R:215]: Ok, changed to "Location Used for the Measurement".

File: data-raw/nv_neuro.R [hunk 10]
```diff
@@ -0,0 +1,248 @@
...
+  # Identifier Variables (Key variables)
+  STUDYID = "Study Identifier",
+  DOMAIN = "Domain Abbreviation",
+  USUBJID = "Unique Subject Identifier",
+  NVSEQ = "Sequence Number",
+  NVLNKID = "Link to Subject's AG Domain Record",
+
+  # Topic Variables\
+  NVTESTCD = "Nervous System Test Short Name",
+  NVTEST = "Nervous System Test Name",
```
[the remark this answers is not in the document]
Comment (jwang-lilly) [reply data-raw/nv_neuro.R:211]: Ok, changed to "Name of Nervous System Test".

File: data-raw/nv_neuro.R [hunk 11]
```diff
@@ -0,0 +1,248 @@
...
+  NVSTRESU = "Standard Units",
+
+  # Timing Variables
+  VISITNUM = "Visit Number",
+  VISIT = "Visit Name",
+  NVDTC = "Date/Time of Collection",
+  NVDY = "Study Day of Collection",
+
+  # Additional Qualifier
+  NVBLFL = "Baseline Flag"
```
[the remark this answers is not in the document]
Comment (jwang-lilly) [reply data-raw/nv_neuro.R:233]: Great idea.  Changed NVBLFL with NVLOBXFL (last observation before exposure flag)

File: data-raw/suppnv_neuro.R (hunk 3 above)
Comment (jwang-lilly) [reply data-raw/suppnv_neuro.R:48]: ok, changed to "Supplemental Qualifiers for Nervous System Findings".

File: data-raw/suppnv_neuro.R (hunk 5 above)
Comment (jwang-lilly) [reply data-raw/suppnv_neuro.R:39]: Ok, changed to "Data Value".

File: data-raw/ag_neuro.R (hunk 4 above)
Comment (jwang-lilly) [reply data-raw/ag_neuro.R:68]: ok, changed to "Procedure Agents"

File: data-raw/suppnv_neuro.R (hunk 6 above)
Comment (jwang-lilly) [reply data-raw/suppnv_neuro.R:40]: Added QORIG and QEVAL to suppnv_neuro.R

File: data-raw/ag_neuro.R [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+  STUDYID = "Study Identifier",
+  DOMAIN = "Domain Abbreviation",
+  USUBJID = "Unique Subject Identifier",
+  AGSEQ = "Sequence Number",
+  AGTRT = "Tracer Name",
+  AGCAT = "Tracer Category",
+  AGDOSE = "Dose per Administration",
+  AGDOSEU = "Dose Units",
+  AGROUTE = "Route of Administration",
+  AGLNKID = "Link to Subject's NV Domain Record",
```
[the remark this answers is not in the document]
Comment (jwang-lilly) [reply data-raw/ag_neuro.R:56]: O, changed to "Link ID".

File: data-raw/ag_neuro.R [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+  )
+
+# Add labels to variables ----
+
+labels <- list(
+  STUDYID = "Study Identifier",
+  DOMAIN = "Domain Abbreviation",
+  USUBJID = "Unique Subject Identifier",
+  AGSEQ = "Sequence Number",
+  AGTRT = "Tracer Name",
```
[the remark this answers is not in the document]
Comment (jwang-lilly) [reply data-raw/ag_neuro.R:51]: Ok changed to "Reported Agent Name".

File: data-raw/ag_neuro.R (hunk 2 above)
Comment (jwang-lilly) [reply data-raw/ag_neuro.R:20]: Thanks Fanny.  Ok.

**Review by Fanny-Gautier — CHANGES_REQUESTED** [review]
Only 1 minor comment (delete commented out line of code) and then it looks good to me. Thanks for all the updates

**Comment by cgpele** [comment]
It's looking good to me! thanks a lot! :)

**Review by manciniedoardo — APPROVED** [review]
